File: commonlib/src/main/java/com/example/commonlib/base/BaseActivity.java
package com.example.commonlib.base;

import android.app.ProgressDialog;
import android.content.pm.ActivityInfo;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.example.commonlib.R;

import butterknife.ButterKnife;
import butterknife.Unbinder;

public class BaseActivity extends AppCompatActivity {

    private Unbinder mBind;
    private ProgressDialog progressDialog;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_base);
        mBind = ButterKnife.bind(this);
        getKeyData();
        initView();
        initData();

        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);

    }

    @Override
    protected void onDestroy() {
        mBind.unbind();
        super.onDestroy();
    }

    protected void getKeyData(){

    }

    protected void initData(){

    }

    protected void initView(){

    }

    public void showProgress(String msg,boolean cancelable){
        progressDialog = ProgressDialog.show(this,null,msg,false,cancelable);
    }

    public void dismissProgress(){
        if (progressDialog == null){
            return;
        }

        if (progressDialog.isShowing()){
            progressDialog.dismiss();
        }
    }

}

File: commonlib/src/main/java/com/hjl/commonlib/base/mvvm/BaseActivity.kt
package com.hjl.commonlib.base.mvvm

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

/**
 * Description //TODO
 * Date 2020/3/2 16:24
 * created by long
 */
abstract class BaseActivity : AppCompatActivity() {

    protected val TAG: String = this.javaClass.simpleName

    lateinit var mContext: BaseActivity

    abstract fun initContentView()

    abstract fun initData()

    abstract fun initView()

    abstract fun initLoad()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initContentView()
        mContext = this

        initData()
        initView()
        initLoad()
    }

}